### PR TITLE
Warn player if soldiers have useless items equipped for non-combat activities

### LINKED ISSUE #479: Clear warning on UISS if no mission can result from this CA
Both Barn Cat and Mister Gruffle made the "mistake" of sending combat gear (like flashbangs) on CAs that do not result in missions (eg. no ambush risk). This is especially easy to do when just starting, when items are most "expensive" since you don't have a stockpile of resources

### PR DESCRIPTION
**Pull request by cannonfodder1**
Closes #479

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc [hunk 1]
```diff
@@ -1500,3 +1500,83 @@ static function UpdateFacilityMissionLocks (optional XComGameState NewGameState)
...
+			if (InfilTemplate != none)
+			{
+				if (InfilTemplate.HoursAdded >= 0 && InfilTemplate.Deterrence <= 0)
+				{
+					return true;
+				}
+			}
+			else
+			{
+				return true;
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc:1548]: You are breaking the loop here (skipping next items) - you sure you want to do that?

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc [hunk 2]
```diff
@@ -1090,6 +1091,64 @@ static protected function EventListenerReturn AddResource_OSS (Object EventData,
...
+    NoteTuple.Data[0].s = Note.Text;
+        
+	NoteTuple.Data[1].kind = LWTVString;
+    NoteTuple.Data[1].s = Note.TextColor;
+        
+	NoteTuple.Data[2].kind = LWTVString;
+    NoteTuple.Data[2].s = Note.BGColor;
+
+    Value.o = NoteTuple;
+    Tuple.Data.AddItem(Value);
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc:1147]: Please fix indentation

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc [hunk 3]
```diff
@@ -1090,6 +1091,64 @@ static protected function EventListenerReturn AddResource_OSS (Object EventData,
...
+	local LWTValue Value;
+	local int i;
+
+	Tuple = LWTuple(EventData);
+	
+	// Check that we are interested in actually doing something
+	if (Tuple == none || Tuple.Id != 'rjSquadSelect_ExtraInfo') return ELR_NoInterrupt;
+	
+	AvengerHUD = `HQPRES.m_kAvengerHUD;
+	ScreenStack = AvengerHUD.Movie.Pres.ScreenStack;
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc:1115]:
```suggestion
	ScreenStack = `SCREENSTACK;
```
Also remove the `local UIAvengerHUD AvengerHUD;`

**Review by Xymanek — CHANGES_REQUESTED** [review]

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc (hunk 1 above)
Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc:1548]: Yes, if the soldier has one irrelevant item then it doesn’t matter what the status of the rest of the items are, so the function returns true.

**Review by Xymanek — APPROVED** [review]

**Pull request merged**